File: dotnet/src/Tyouqu.Database.Abstractions/SqlTemplateOptions.cs
namespace Tyouqu.Database.Abstractions;

public sealed class SqlTemplateOptions
{
    public string RootPath { get; set; } = "sql";

    public bool EnableHotReload { get; set; }

    public string ReloadMode { get; set; } = "Manual";

    public bool FailOnDuplicateSqlId { get; set; } = true;

    public bool FailOnMissingSqlId { get; set; } = true;
}

File: java/src/main/java/com/tyouqu/database/SqlTemplateOptions.java
package com.tyouqu.database;

public record SqlTemplateOptions(
    String rootPath,
    boolean failOnDuplicateSqlId,
    boolean failOnMissingSqlId
) {
    public SqlTemplateOptions {
        rootPath = rootPath == null || rootPath.isBlank() ? "sql" : rootPath;
    }

    public static Builder builder() {
        return new Builder();
    }

    public static final class Builder {
        private String rootPath = "sql";
        private boolean failOnDuplicateSqlId = true;
        private boolean failOnMissingSqlId = true;

        public Builder rootPath(String rootPath) {
            this.rootPath = rootPath;
            return this;
        }

        public Builder failOnDuplicateSqlId(boolean failOnDuplicateSqlId) {
            this.failOnDuplicateSqlId = failOnDuplicateSqlId;
            return this;
        }

        public Builder failOnMissingSqlId(boolean failOnMissingSqlId) {
            this.failOnMissingSqlId = failOnMissingSqlId;
            return this;
        }

        public SqlTemplateOptions build() {
            return new SqlTemplateOptions(rootPath, failOnDuplicateSqlId, failOnMissingSqlId);
        }
    }
}
